server/cmd/main.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/skip-mev/ironbird/db"
+	"github.com/skip-mev/ironbird/server"
+	"github.com/skip-mev/ironbird/types"
+	"go.uber.org/zap"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	_ "github.com/caddyserver/caddy/v2/modules/standard"
+	"google.golang.org/grpc/grpclog"
+)
+
+type CaddyModule struct {
+	server *server.IronbirdServer
+	config types.TemporalConfig
+}
+
+func (CaddyModule) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID: "http.handlers.ironbird",
+		New: func() caddy.Module {
+			m := &CaddyModule{
+				config: types.TemporalConfig{
+					Host:      "127.0.0.1:7233", // TODO(nadim-az): update when deploying to prod to point to temporalconfig
+					Namespace: "default",
+				},
+			}
+			return m
+		},
+	}
+}
+
+func (m *CaddyModule) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	return nil
+}
+
+func (m *CaddyModule) Provision(ctx caddy.Context) error {
+	var err error
+
+	dbPath := getEnvOrDefault("DATABASE_PATH", "./ironbird.db")
+
+	database, err := db.NewSQLiteDB(dbPath)
+	if err != nil {
+		return fmt.Errorf("failed to connect to database: %w", err)
+	}
+
+	logger, _ := zap.NewDevelopment()
+
+	m.server, err = server.NewIronbirdServer(m.config, database, logger)
+	return err
+}
+
+func (m *CaddyModule) Cleanup() error {
+	if m.server != nil {
+		return m.server.Close()
+	}
+	return nil
+}
+
+// handleIronbirdRoutes routes HTTP requests to the appropriate handler based on method and path.
+// Returns true if the request was handled, false otherwise.
+func handleIronbirdRoutes(w http.ResponseWriter, r *http.Request, server *server.IronbirdServer) bool {
+	const (
+		basePath      = "/ironbird"
+		workflowPath  = basePath + "/workflow"
+		workflowsPath = basePath + "/workflows"
+		loadTestPath  = basePath + "/loadtest/"
+	)
+
+	handleRequest := func(handlerFunc func(http.ResponseWriter, *http.Request) error) bool {
+		if err := handlerFunc(w, r); err != nil {
+			errMsg := fmt.Sprintf("Error: %v", err)
+			http.Error(w, errMsg, http.StatusInternalServerError)
+			return false
+		}
+		return true
+	}
+
+	switch {
+	// POST /ironbird/workflow/{id}/cancel - Cancel a workflow
```

Comment:
why did you choose caddy for the backend? i don't think it's a big deal now, but i also don't think it was the best choice compared to what we use as a standard - gRPC + gRPC-gateway